eruditus/lib/discord_util.py

```diff
@@ -0,0 +1,246 @@
+import re
+from datetime import datetime
+from typing import Optional
+
+import aiohttp
+import discord
+
+from config import (
+    CHALLENGE_COLLECTION,
+    CTF_COLLECTION,
+    DBNAME,
+    MAX_CONTENT_SIZE,
+    MONGO,
+    TEAM_NAME,
+)
+from lib.platforms import PlatformCTX, match_platform
+from lib.util import plot_scoreboard
+
+
+async def get_ctf_info(
+    interaction: discord.Interaction, name: Optional[str] = None
+) -> Optional[dict]:
+    # Trying to find ctf by channel category
+    if name is None:
+        ctf = MONGO[DBNAME][CTF_COLLECTION].find_one(
+            {"guild_category": interaction.channel.category_id}
+        )
+
+        if ctf is None:
+            await interaction.followup.send(
+                (
+                    "Run this command from within a CTF channel, or provide the "
+                    "name of the CTF you wish to delete."
+                )
+            )
+            return None
+
+        return ctf
+
+    # Trying to find ctf by provided name
+    ctf = MONGO[DBNAME][f"{CTF_COLLECTION}"].find_one(
+        {"name": re.compile(f"^{re.escape(name.strip())}$", re.IGNORECASE)}
+    )
+    if ctf is None:
+        await interaction.followup.send("No such CTF.")
+        return None
+
+    return ctf
+
+
+async def get_challenge_solvers(
+    interaction: discord.Interaction, challenge: dict, members: Optional[str] = None
+) -> list[str]:
+    if interaction.user.name not in challenge["players"]:
+        challenge["players"].append(interaction.user.name)
+
+    result: list[str] = [interaction.user.name]
+
+    if members is None:
+        return result
+
+    for mention in re.split(r"<@!?([0-9]{15,20})>", members):
+        if mention == "":
+            continue
+
+        # If id and we can fetch the user
+        if re.match(r"^[0-9]{15,20}$", mention):
+            member_info = await interaction.guild.fetch_member(int(mention))
+
+            if member_info:
+                result.append(member_info.name)
+                continue
+
+        # Otherwise, adding as a plain text
+        result.extend(mention.split())
+
+    return result
```

Comment:
```suggestion
    return [interaction.user.name] + (
        []
        if members is None
        else [
            member.name
            for member_id in re.findall(r"<@!?([0-9]{15,20})>", members)
            if (member := await interaction.guild.fetch_member(int(member_id)))
        ]
    )
```
It's up to the user to properly tag the other members who collaborated in the challenge, it's even simpler than typing names manually so I don't think handling the plaintext user names is necessary.

Reply:
Not that i don't agree with you, but even some of my team's members are sometimes forgetting that you need to actually tag people. There are also some players that could not be in the server, but still helped with challenge.

Reply:
It doesn't have a huge impact still, I think we should skip it.

Reply:
Sure, as you wish